contracts/staking/ValidatorManager.sol

```diff
@@ -108,10 +105,108 @@ abstract contract ValidatorManager is
         );
         $._maximumChurnPercentage = settings.maximumChurnPercentage;
         $._churnPeriodSeconds = settings.churnPeriodSeconds;
+    }
+
+    modifier initializedValidatorSet() {
+        require(
+            _getValidatorManagerStorage()._initializedValidatorSet,
+            "ValidatorManager: validator set not initialized"
+        );
+        _;
+    }
+
+    /**
+     * @notice See {IValidatorManager-initializeValidatorSet}.
+     */
+    function initializeValidatorSet(
+        SubnetConversionData calldata subnetConversionData,
+        uint32 messageIndex
+    ) external {
+        ValidatorManagerStorage storage $ = _getValidatorManagerStorage();
+        require(!$._initializedValidatorSet, "ValidatorManager: already initialized validator set");
+        require(
+            subnetConversionData.blockchainID == WARP_MESSENGER.getBlockchainID(),
+            "ValidatorManager: invalid blockchain ID"
+        );
+        require(
+            address(subnetConversionData.validatorManagerAddress) == address(this),
+            "ValidatorManager: invalid validator manager address"
+        );
+        require(
+            subnetConversionData.initialValidators.length > 0,
+            "ValidatorManager: no initial validators"
+        );
+        uint256 numInitialValidators = subnetConversionData.initialValidators.length;
+        require(
+            numInitialValidators < type(uint32).max, "ValidatorManager: too many initial validators"
+        );
```

Comment:
Do we really have to check that we haven't included over 4 billion validators?

Reply:
IMO this can be removed along with the other checks mentioned [here](https://github.com/ava-labs/teleporter/pull/547#discussion_r1764103545), as long as we assume the P-Chain will enforce sane initial validator set size limits